pkg/operator/configobserver/node/observe_latency_profile.go

```diff
@@ -0,0 +1,138 @@
+package node
+
+import (
+	"reflect"
+	"strings"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// LatencyProfileArgVal is used to set arg value pairs for each worker latency profile
+type LatencyProfileArgVal struct {
+	// path to required argument in observed config
+	// eg. observedConfig.extendedArguments.node-monitor-grace-period
+	// is []string{"extendedArguments", "node-monitor-grace-period"}
+	ArgConfigNestedPath []string
+	// value that is to be set for the given argument
+	ArgValue string
+}
+
+// CreateLatencyProfileObserverFn is used to create ObserveConfigFunc that can be used with an configobservation controller to trigger
+// changes to different arg val pairs in observedConfig.* fields and update them on the basis of current worker latency profile
+func CreateLatencyProfileObserverFn(latencyProfileValMap map[configv1.WorkerLatencyProfileType][]LatencyProfileArgVal, conditionFn func() (bool, error)) configobserver.ObserveConfigFunc {
+	observeLatencyProfileFn := func(
+		genericListers configobserver.Listers,
+		eventRecorder events.Recorder,
+		existingConfig map[string]interface{},
+	) (ret map[string]interface{}, errs []error) {
+
+		// traverse among different worker latency profiles and their required arg val pairs
+		// and obtain a unique list of paths that are to be used with the observedConfig
+		uniqueNestedPaths := getUniqueNestedPaths(latencyProfileValMap)
+
+		defer func() {
+			// Prune the observed config so that it only contains fields specific to this observer
+			ret = configobserver.Pruned(ret, uniqueNestedPaths...)
+
+			// in case of existingConfig and currently observedConfig has mismatch in fields that are
+			// being monitored eg. apiServerArguments.default-not-ready-toleration-seconds,
+			// extendedArguments.node-monitor-grace-period, etc.
+			prevRet := configobserver.Pruned(existingConfig, uniqueNestedPaths...)
+			if !reflect.DeepEqual(prevRet, ret) {
+				eventRecorder.Eventf("LatencyProfileUpdated", "latency profile on cluster was updated, observed-config = %v", ret)
```

Comment:
I actually added it in favour of https://github.com/openshift/cluster-kube-controller-manager-operator/pull/611#discussion_r859554873.

Also, if you check the https://github.com/openshift/library-go/blob/5e6227194fbeb42769809e7026625dd1190d4ab4/pkg/operator/configobserver/apiserver/observe_tlssecurityprofile.go#L69-L71 they're firing a similar event (similarly as an extra to the `ObservedConfigChanged` event as you mentioned).

It might be helpful as well for the users to know when the respective operator has enacted upon the newly applied latency profile.

Reply:
I would vote for simplifying it if we have one event already. And if the users want to react to these changes they can always watch the conditions which is better API wise I think.

Reply:
Done, removed the extra event.